### DISCUSSION TITLE
Raft options

### DISCUSSION THREAD

**Discussion by hamidreza-hosseini** [seaweedfs/seaweedfs#3072] [Q&A] [answered]
Hi, @chrislusf @kmlebedev 
I was reading Seaweedfs issues and "weed master --help" and I encountered raft options but help explanation wasn't enough to understand them
1. What are these two options and when should we enable and use them?

```
  -raftBootstrap
    	Whether to bootstrap the Raft cluster
  -raftHashicorp
    	use hashicorp raft
```
The following config is what I use in seaweed-master servers to be run (e.g: I use "resumeState=true" for a problem that @kmlebedev suggested  )

```

ExecStart=/usr/local/bin/weed -logtostderr=true -alsologtostderr=false -v=0 master -mdir="{{ master_directory }}" -ip={{ ansible_facts['hostname'] }}.local -port=9333 -peers="{{ Master_servers }}" -defaultReplication=010 -volumeSizeLimitMB={{ volume_limit_size }} -resumeState=true -metrics.address={{ prometheus_push_server }}
```

Thanks in advance

**Comment by kmlebedev** [comment, marked as answer]
@kaaaq Hi, option raftHashicorp switches to a more mature raft implementation from Hashicorp
The raft Bootstrap option, if necessary, allows you to force the launch of a raft from Hashicorp without a quorum, but with the removal of the state(max volume Id)

Using the resumeState option is dangerous without the raftHashicorp option

**Reply by hamidreza-hosseini** [reply]
> @kaaaq Hi, option raftHashicorp switches to a more mature raft implementation from Hashicorp The raft Bootstrap option, if necessary, allows you to force the launch of a raft from Hashicorp without a quorum, but with the removal of the state(max volume Id)

Hi @kmlebedev ,
1. Thanks for your answer, but still raftBootstrap option is not totally clear to me, could you please explain it more?
2. What would be the negative aspect of "removal of the state(max volume Id)" on my cluster? 
3. raftHashicorp implementation in Seaweedfs is stable enough to use?

**Reply by kmlebedev** [reply]
1.Forced re-initialization of the cluster from a clean slate
2. In the growth race of new volumes, the same ID is allocated for different collections
3. Yes
